### DISCUSSION TITLE
Help with runtime error at MapNode::get()

### DISCUSSION THREAD

**Discussion by AhmedAredah** [pelicanmapping/osgearth#2474] [Support Q&A] [answered]
I am trying to read an earth file and visualize it. The file reading went good however, when I pass it to MapNode::get() it throws an error. Below is the portion of the code that loads the file:
  
 ```
 
    // read in the Earth file:
    osg::ref_ptr<osg::Node> node  = osgDB::readRefNodeFile(filename);
    if (!node .valid())
    {
        qDebug() << "Failed to load Earth model from: " <<
            QString::fromStdString(filename);
        return;
    }
    
    osg::Node* n = node.get();
    osg::ref_ptr<MapNode> mapNode = MapNode::get(n);          << error in insertion here. 
    if (!mapNode.valid())
    {
        qDebug() << "Loaded scene graph does not contain a MapNode";
        return;
    }
```

Error: The inferior stopped because it triggered an exception.
```
Debug Assertion Failed!
Program: C:\Program Files\OpenSceneGraph\bin\osg202-osgd.dll
File: Files (x86)\Microsoft Visual
Studio\2019\Professional\VC\Tools\MSVC\14.29.30133\include\vector
Line: 805
Expression: vector emplace iterator outside range
For information on how your program can cause an assertion
failure, see the Visual C++ documentation on asserts.
```

The error happens at this line (subprocess of the MapNode::get() function): 
`        inline void pushOntoNodePath(Node* node) { if (_traversalMode!=TRAVERSE_PARENTS) _nodePath.push_back(node); else _nodePath.insert(_nodePath.begin(),node); }`

The node is being inserted at the beginning of the vector but I have no clue why it is throwing that error. everything seems fine to me.

Note: I am loading one of the test earth files (glsl.earth)

Env: Windows 11 x64 bit.

**Comment by gwaldron** [comment, marked as answer]
This error is occuring inside OpenSceneGraph. I suspect you have a bad build. Maybe you are mixing versions in your path? Or mixing debug and release builds?

**Reply by AhmedAredah** [reply]
Oh, that makes sense. Thank you for helping out and for your patience. I do appreciate it.

I was passing ${osgEarth_LIBRARIES} in the cmake file and didn't pay attention to the referred file.
